src/main/java/software/amazon/cloudformation/exceptions/CfnNotStabilizedException.java

```diff
@@ -25,14 +25,62 @@ public CfnNotStabilizedException(final Throwable cause) {
         super(cause, ERROR_CODE);
     }
 
+    /**
+     * @param resourceTypeName
+     * @param resourceIdentifier
+     * @param reason Reason why the resource did not stabilize. This should include the current and
+     *               desired state.
+     *
+     *               Example: "Exceeded retry limit for DescribeResourceStatus.
+     *               Current Value: IN_PROGRESS. Desired Value: COMPLETE."
```

Comment:
nice, I like the docs and providing an example.  Couple comments:
* I don't think desired state is necessary
* For AWS resources, it should include the current state, Api name in iam format (like ec2:DescribeInstances) and the RequestId for that API (for non AWS resources it would be whatever the equivalent debugging info)

The example here is also a little bit confusing. Is this one example or two examples?  Here's one

```
* Instance Status: pending (Api: ec2:DescribeInstances, RequestId: abcd-efgh-ijkl)
* Current Status: modifying (Api: rds:DescribeDBClusters, RequestId: abcd-efgh-ijkl)
```

Another option here would be to take the apiName and requestid as parameters